src/position.cpp

```diff
@@ -579,6 +579,7 @@ bool Position::legal(Move m) const {
 /// Position::pseudo_legal() takes a random move and tests whether the move is
 /// pseudo legal. It is used to validate moves from TT that can be corrupted
 /// due to SMP concurrent access or hash position key aliasing.
+/// MOVE_NONE is represented as SQ_A1 to SQ_A1 which is never pseudo_legal.
 
```

Comment:
should be 'MOVE_NONE is never pseudo_legal' without the details. 

Otherwise looks good to me.

Reply:
Just saying that MOVE_NONE is never pseudo legal seems confusing to me.

The correctness of the code depends on the fact that MOVE_NONE is _implemented_ by a move which is never pseudo legal. So it seems to me that at least this much should be mentioned (as is done in the current comment).